addon/helpers/transition-to.js

```diff
@@ -5,13 +5,17 @@ export default Helper.extend({
   router: computed(function() {
     return getOwner(this).lookup('router:main');
   }).readOnly(),
+
   compute([routeName, ...params]) {
-    let router = get(this, 'router');
+    const router = get(this, 'router');
     assert('[ember-transition-helper] Unable to lookup router', router);
+
     return function(...invocationArgs) {
-      let args = params.concat(invocationArgs);
-      let transitionArgs = params.length ? [routeName, ...params] : [routeName];
-      router.transitionTo.apply(router, transitionArgs);
+      const args = params.concat(invocationArgs);
+      const transitionArgs = args.length ? [routeName, ...args] : [routeName];
```

Comment:
Is `args` vs `params` the only functional change here? I decided to re-make this addon, both for practice and for the sake of keeping things up-to-date to reduce effects of code rot, and would like to incorporate your work here into my [`jj-abrams` PR](https://github.com/peec/ember-transition-helper/pull/14)

Reply:
@jacobq thanks for resurrecting this! Yep, changing `params` out for `args` was the only change needed at the time (and for my use case at that point - we've since stopped using this helper).

It looks like with these changes on your branch, the tests fail because the `invocationArgs` include potentially one or more additional parameters depending on how the helper is used:

```hbs
<a onclick={{transition-to "test" "some-arg"}}>Usage 1</a>
<a {{action (transition-to "test" "some-arg")}}>Usage 2</a>
```

`Usage 1` breaks because an event object gets passed to the helper when it's invoked, as well as the `"test"` and `"some-arg"` params. `Usage 2` is fine though.

`Usage 1` also breaks with the solution we now use in our own app, but we've never known because we only ever use either `Usage 2`-type syntax or a custom button component which never passes the event object through to the invoked action.

The "ultimate" fix-all to make this helper usable in all action forms would be to ignore those extra parameters, but I'm not sure how easy it is to reliably detect whether that extra event param is present (and actually in another quick test I tried, I found that I got _two_ extra params - the target element and the event object - which makes the matter even harder).

All that being said, I can see two potential paths from here:

1. Leave things as they are and don't support currying.
2. Make my change to support currying, but break some existing usages of the helper that don't use an action "syntax" that works 😬 

I'd be interested to hear your thoughts on if/how we proceed from here! 👍